packages/code-studio/src/settings/ThemeSectionContent.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useCallback } from 'react';
+import {
+  Item,
+  ItemKey,
+  Picker,
+  ThemePicker,
+  useTheme,
+} from '@deephaven/components';
+import { assertNotNull } from '@deephaven/utils';
+import { useDispatch, useSelector } from 'react-redux';
+import { getSettings, updateSettings, type RootState } from '@deephaven/redux';
+
+export function ThemeSectionContent(): JSX.Element {
+  const theme = useTheme();
+  const settings = useSelector<RootState, ReturnType<typeof getSettings>>(
+    getSettings
+  );
+  const dispatch = useDispatch();
+
+  const updateDensity = useCallback(
+    (density: ItemKey | null) => {
+      if (
+        density !== 'normal' &&
+        density !== 'compact' &&
+        density !== 'spacious'
+      ) {
+        throw new Error(`Invalid grid density value: ${density}`);
+      }
+      dispatch(updateSettings({ gridDensity: density }));
+    },
+    [dispatch]
+  );
+
+  const density = settings.gridDensity ?? 'normal';
+
+  assertNotNull(theme, 'ThemeContext value is null');
+
+  return (
+    <>
+      <ThemePicker />
+      <Picker
+        label="Choose grid density"
```

Comment:
Maybe a question for @dsmmcken, but should this density apply to other Spectrum components such as ListView by default as well? https://react-spectrum.adobe.com/react-spectrum/ListView.html#density
Though some items (like `Tabs`) don't have a `spacious` option.

Reply:
One vote for no is users of dh.ui should already be able to specify `density` on those components.

In our UI, we would then need to test anywhere we use list or tabs (or others w/ density) looks ok w/ all density options

Reply:
Yup limit the scope of this ticket, but could be interesting to do later.